README.md

```diff
@@ -89,38 +89,68 @@ Note that memory usage is higher when executing a query with parallelism greater
+RangeQuery/at_modifier_with_positive_offset_vector-8    27.1MB ± 0%    34.0MB ± 0%   +25.33%  (p=0.000 n=10+10)
+RangeQuery/clamp-8                                      31.1MB ± 0%    38.0MB ± 0%   +22.02%  (p=0.000 n=10+10)
+RangeQuery/clamp_min-8                                  31.1MB ± 0%    37.9MB ± 0%   +21.83%  (p=0.000 n=10+10)
+RangeQuery/complex_func_query-8                         45.6MB ± 0%    41.9MB ± 0%    -8.21%  (p=0.000 n=9+9)
+RangeQuery/func_within_func_query-8                     32.1MB ± 0%    39.4MB ± 0%   +22.71%  (p=0.000 n=10+10)
+RangeQuery/aggr_within_func_query-8                     32.1MB ± 0%    39.3MB ± 0%   +22.55%  (p=0.000 n=10+10)
+
+name                                                  old allocs/op  new allocs/op  delta
+RangeQuery/vector_selector-8                             69.1k ± 0%     78.2k ± 0%   +13.07%  (p=0.000 n=10+10)
+RangeQuery/sum-8                                         73.8k ± 0%     72.5k ± 0%    -1.80%  (p=0.000 n=9+10)
+RangeQuery/sum_by_pod-8                                   569k ± 0%      158k ± 0%   -72.28%  (p=0.000 n=9+10)
+RangeQuery/rate-8                                        78.2k ± 0%     99.0k ± 0%   +26.60%  (p=0.000 n=9+9)
+RangeQuery/sum_rate-8                                    82.9k ± 0%     93.6k ± 0%   +12.88%  (p=0.000 n=9+10)
+RangeQuery/sum_by_rate-8                                  576k ± 0%      179k ± 0%   -68.98%  (p=0.000 n=10+9)
+RangeQuery/binary_operation_with_one_to_one-8            55.3k ± 0%     68.6k ± 0%   +24.05%  (p=0.000 n=10+10)
+RangeQuery/binary_operation_with_many_to_one-8            594k ± 0%      129k ± 0%   -78.34%  (p=0.000 n=10+10)
+RangeQuery/binary_operation_with_vector_and_scalar-8     84.4k ± 0%     91.9k ± 0%    +8.96%  (p=0.000 n=10+10)
+RangeQuery/unary_negation-8                              78.1k ± 0%     90.3k ± 0%   +15.59%  (p=0.000 n=10+10)
+RangeQuery/vector_and_scalar_comparison-8                75.3k ± 0%     91.9k ± 0%   +22.09%  (p=0.000 n=10+9)
+RangeQuery/positive_offset_vector-8                      57.1k ± 0%     66.1k ± 0%   +15.69%  (p=0.000 n=10+10)
+RangeQuery/at_modifier_-8                                75.1k ± 0%     58.7k ± 0%   -21.85%  (p=0.000 n=10+9)
+RangeQuery/at_modifier_with_positive_offset_vector-8     69.1k ± 0%     52.7k ± 0%   -23.76%  (p=0.000 n=8+10)
+RangeQuery/clamp-8                                       84.7k ± 0%    803.2k ± 0%  +848.76%  (p=0.000 n=10+10)
+RangeQuery/clamp_min-8                                   84.4k ± 0%    802.2k ± 0%  +850.57%  (p=0.000 n=10+9)
+RangeQuery/complex_func_query-8                           100k ± 0%      817k ± 0%  +717.42%  (p=0.000 n=10+9)
+RangeQuery/func_within_func_query-8                      93.7k ± 0%    818.3k ± 0%  +772.88%  (p=0.000 n=10+10)
+RangeQuery/aggr_within_func_query-8                      93.7k ± 0%    818.3k ± 0%  +772.81%  (p=0.000 n=10+9)
```

Comment:
Probably let's have an issue to track the allocation regression. 10x seems not normal, maybe we could have some kind of pooling.

Reply:
Agree, I also noticed this which is strange. But total bytes used seems to be in line which is why I was not concerned too much.

Reply:
Yup! Will create an issue. There isn't a memory/cpu regression but allocs seem abnormally high